hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/HoodieIndex.java

```diff
@@ -60,7 +57,7 @@ protected HoodieIndex(HoodieWriteConfig config) {
   @Deprecated
   @PublicAPIMethod(maturity = ApiMaturityLevel.DEPRECATED)
   public I tagLocation(I records, HoodieEngineContext context,
-                       HoodieTable<T, I, K, O> hoodieTable) throws HoodieIndexException {
+                       HoodieTable hoodieTable) throws HoodieIndexException {
```

Comment:
As general rule of thumb, i think we should aim to fix all raw-types occurrences in the code-base.

Can you please fix new instances so that we at least don't add to our debt?

Reply:
@alexeykudinkin Yes, i plan to make a follow up PR to remove T from `HoodieTable`, then i can make the interface taking `HoodieTable<I, K, O>`.

Reply:
The deprecated APIs are going to be removed in a separate PR to keep the scope of this PR limited.